src/pybind/ceph_volume_client.py

```diff
@@ -439,20 +441,27 @@ def create_volume(self, volume_path, size=None, data_isolated=False):
         path = self._get_path(volume_path)
 
         self._mkdir_p(path)
-
         if size is not None:
             self.fs.setxattr(path, 'ceph.quota.max_bytes', size.__str__(), 0)
 
-        if data_isolated:
-            pool_name = "{0}{1}".format(self.POOL_PREFIX, volume_path.volume_id)
-            pool_id = self._create_volume_pool(pool_name)
-            mds_map = self._rados_command("mds dump", {})
-            if pool_id not in mds_map['data_pools']:
-                self._rados_command("mds add_data_pool", {
-                    'pool': pool_name
-                })
-            self.fs.setxattr(path, 'ceph.dir.layout.pool', pool_name, 0)
-
+        namespace = "{0}{1}".format(self.POOL_NS_PREFIX, volume_path.volume_id)
+        log.info("create_volume: {0}, using rados namespace {1} to isolate data.".format(volume_path, namespace))
+        try:
+            self.fs.setxattr(path, 'ceph.dir.layout.pool_namespace', namespace, 0)
```

Comment:
It's confusing that namespace is used regardless 'data_isolated' flag.  If the flag was meant to represent whether volume should be created in a way that access to it can be controlled separately, then this flag should be honored.